FastSimulation/CTPPSFastGeometry/src/CTPPSToFDetector.cc

```diff
@@ -81,27 +82,21 @@ void CTPPSToFDetector::AddHit(double x, double y, double tof) {
   nADC_[cellid].push_back(1);
 }
 int CTPPSToFDetector::findCellId(double x, double y) {
-  int y_idx, x_idx;
-  // first, get the row number
-  unsigned int i;
-  unsigned int start_idx = 0;
-  unsigned int end_idx = cellRow_.size();
-  for (i = 0; i < cellRow_.size(); i++) {
-    if (y >= cellRow_.at(i).first && y <= cellRow_.at(i).second)
-      break;
-  }
-  if (i >= cellRow_.size())
+  auto it = std::find_if(
```

Comment:
Values of start_idx and end_idx were not used ([overwritten later](https://github.com/cms-sw/cmssw/pull/46255/files#diff-17d81d12169ae03f5dcddf62becdc9b805803687af260d7eca6d6d4464475ce0L96-L97)).

Reply:
The algorithm should be understood in general.